api/src/main/java/org/openmrs/module/fhir2/api/dao/impl/BaseEncounterDao.java

```diff
@@ -46,10 +57,70 @@ protected void setupSearchParams(Criteria criteria, SearchParameterMap theParams
 				case FhirConstants.COMMON_SEARCH_HANDLER:
 					handleCommonSearchParameters(entry.getValue()).ifPresent(criteria::add);
 					break;
+				case FhirConstants.HAS_SEARCH_HANDLER:
+					entry.getValue().forEach(param -> handleHasAndListParam(criteria, (HasAndListParam) param.getParam()));
+					break;
 			}
 		});
 	}
 	
+	/**
+	 * Handle _has parameters that are passed in to constrain the Encounter resource on properties of
+	 * dependent resources
+	 */
+	protected void handleHasAndListParam(Criteria criteria, HasAndListParam hasAndListParam) {
+		if (hasAndListParam != null) {
+			log.debug("Handling hasAndListParam");
+			hasAndListParam.getValuesAsQueryTokens().forEach(hasOrListParam -> {
+				if (!hasOrListParam.getValuesAsQueryTokens().isEmpty()) {
+					
+					log.debug("Handling hasOrListParam");
+					// Making the assumption that any "orListParams" match everything except for the value
+					HasParam hasParam = hasOrListParam.getValuesAsQueryTokens().get(0);
+					Set<String> values = new HashSet<>();
+					hasOrListParam.getValuesAsQueryTokens().forEach(orParam -> values.add(orParam.getParameterValue()));
+					
+					log.debug("Handling hasParam = " + hasParam.getQueryParameterQualifier());
+					log.debug("With value in " + values);
+					
+					boolean handled = false;
+					
+					// Support constraining encounter resources to those that contain only certain Medication Requests
+					if (FhirConstants.MEDICATION_REQUEST.equals(hasParam.getTargetResourceType())) {
+						if (MedicationRequest.SP_ENCOUNTER.equals(hasParam.getReferenceFieldName())) {
```

Comment:
This will effectively add the inner join restriction, and only return encounters that have associated drug orders at this point, even if there are no valid/matching parameter values found.  If this isn't ok, we can adapt, but this was simplest.